odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,546 @@
+    displacement : `ProductSpace` element
+        The vector field (displacement field) used in the linearized
+        deformation.
+
+    Returns
+    -------
+    deformed_template : `numpy.ndarray`
+        Deformed template as an numpy array.
+
+    Examples
+    --------
+    >>> import odl
+    >>> space = odl.uniform_discr(0, 1, 5)
+    >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+    >>> template = space.element([0, 0, 1, 0, 0])
+    >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+    >>> linear_deform(template, displacement_field)
+    array([ 0.,  0.,  1.,  1.,  0.])
+    """
+    image_pts = template.space.grid.points()
+    image_pts += np.asarray(displacement).T
+    return template.interpolation(image_pts.T, bounds_check=False)
+
+
+def deform_grad(grad_f, displacement):
+    """Compute the deformation of the template gradient.
+
+    The template gradient ``grad(I)`` after deformation with displacement
+    field ``v``:
+
+        grad(I)(Id + v)
+
+    where ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain.
+
+    Parameters
+    ----------
+    grad_f: `ProductSpace` element
+        Gradient of the template, i.e. a vector field on the
+        image domain
+    displacement : `ProductSpace` element
+        The vector field (displacement field) used in the linearized
+        deformation.
+    Examples
+    --------
+    >>> import odl
+    >>> space = odl.uniform_discr(0, 1, 5)
+    >>> template = space.element([0, 0, 1, 0, 0])
+    >>> grad_op = odl.Gradient(template.space, method='forward')
+    >>> template_grad = grad_op(template)
+    >>> template_grad
+    ProductSpace(uniform_discr(0.0, 1.0, 5), 1).element([
+        [0.0, 5.0, -5.0, 0.0, 0.0]
+    ])
+    >>> disp_field_space = grad_op.range
+    >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+    >>> deform_grad(template_grad, displacement_field)
+    ProductSpace(uniform_discr(0.0, 1.0, 5), 1).element([
+        [0.0, 5.0, -5.0, -5.0, 0.0]
+    ])
+    """
+    return displacement.space.element(
+        [linear_deform(gf, displacement) for gf in grad_f])
+
+
+class LinDeformFixedTempl(Operator):
+
+    """Operator mapping a displacement field to corresponding deformed template.
+
+    The operator maps a displacement field to the corresponding
+    deformed fixed template ``I``:
+
+        v --> I(Id + v)
+
+    where ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain.
+    """
+
+    def __init__(self, template):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        template : `DiscreteLp` element
+            Fixed template that is to be deformed by the displacement field.
```

Comment:
First: This description is good.
In the light of [this comment](https://github.com/odlgroup/odl/pull/448/files/d71a973d6a397894e0d3b20fcb979bacda74d756#r69896533), change the doc to
```
template : `DiscreteLpVector` or array-like
    Fixed template that is to be deformed.
    If ``domain`` is not given, ``template`` must be a `DiscreteLpVector`,
    and ``template.space`` is used as domain of this operator.
    If ``domain`` is given, ``template`` can be anything that is understood
    by the ``domain.element()`` method.
domain : `DiscreteLp`, optional
    Space of templates on which this operator acts, i.e. the operator
    domain. If not given, ``template.space`` is used as domain.
```